thetanuts/thetanuts/config.py

```diff
@@ -0,0 +1,277 @@
+import json
+import time
+from typing import Any
+
+import web3
+from web3.middleware import geth_poa_middleware
+from web3.types import Nonce
+
+from sdk_commons.chains import Chains
+from sdk_commons.config import BidValidation, OfferDetails, OfferTokenDetails, SDKConfig
+from thetanuts.definitions import Bid
+from thetanuts.wallet import Wallet
+
+
+class AuthorizationPages:
+    mainnet = "https://thetanuts.finance/paradigm/mm-approval"
+    polygon = "https://thetanuts.finance/paradigm/mm-matic-approval"
+    testnet = "https://thetanuts.finance/paradigm/mm-matic-approval"
+
+
+class Thetanuts(SDKConfig):
+
+    authorization_pages = AuthorizationPages
+    supported_chains = [Chains.ETHEREUM, Chains.MATIC]
+
+    def create_offer(
+        self,
+        *,
+        contract_address: str,
+        chain_id: int,
+        rpc_uri: str,
+        oToken: str,
+        bidding_token: str,
+        min_price: int,
+        min_bid_size: int,
+        offer_amount: int,
+        public_key: str,
+        private_key: str,
+        **kwargs: Any,
+    ) -> str:
+        """
+        Start new round by forcefully ending previous round
+        Needs to be done by Vault Owner
+        """
+
+        w3 = web3.Web3(web3.HTTPProvider(rpc_uri))
+        if chain_id == Chains.MATIC.value:
+            w3.middleware_onion.inject(geth_poa_middleware, layer=0)
+        vaultContract = w3.eth.contract(
+            w3.toChecksumAddress(oToken),
+            # w3.toChecksumAddress(vault_contract_address),
+            abi=json.load(open("thetanuts/abis/Vault.json", "r")),
+        )
+
+        nonce = w3.eth.get_transaction_count(public_key)
+
+        if vaultContract.functions.expiry().call() > 0:  # Round in progress, let's end it
+            currentTime = int(time.time())
+            tx = w3.eth.send_raw_transaction(
+                w3.eth.account.sign_transaction(
+                    vaultContract.functions.setExpiry(currentTime).build_transaction(
+                        {'nonce': Nonce(nonce), 'from': public_key}
+                    ),
+                    private_key,
+                ).rawTransaction
+            )
+            print("Sent OWNER transaction for setting expiry", tx.hex())
+            w3.eth.wait_for_transaction_receipt(tx)
+            time.sleep(5)
```

Comment:
5 is a safe value to be sure to complete the transaction or has some exrta special meaning?
Could be reduced or 5 is the minimum value to be considered as safe as a wait time?

Reply:
Of course sleeps should be avoided... but since create_offer is only used for testing purpose, I think can be ok here

Reply:
yea 5 seconds just a arbitrary number, wait_for_transaction_receipt should pend until the tx is complete, but i seem to need to add some propagation time for the various RPCs to be sure. but yes it is only for testing!